src/ActionBar/ActionBar.js

```diff
@@ -1,42 +1,121 @@
         }
     }, []);
 
+    const actionBarHeaderClasses = classnames(
+        'fd-action-bar__header',
+        className
+    );
+
+    const actionBarBackClasses = classnames(
+        'fd-action-bar__back',
+        buttonContainerClassNames
+    );
+
+    const actionBarDescriptionClasses = classnames(
+        'fd-action-bar__description',
+        {
+            'fd-action-bar__description--back': onClick
+        }
+    );
+
+    const actionBarActionsClasses = classnames(
+        'fd-action-bar__actions',
+        actionClassNames
+    );
+
+    const HeadingTag = `h${headingLevel}`;
+
     return (
         <div {...props}
             className={actionBarClasses}
             ref={ref}>
-            {children}
+            <div {...props} className={actionBarHeaderClasses}>
+                {onClick && (<div className={actionBarBackClasses}>
+                    <Button
+                        {...buttonProps}
+                        compact
+                        disableStyles={disableStyles}
+                        glyph='navigation-left-arrow'
+                        onClick={onClick}
+                        option='transparent' />
+                </div>)}
+                <HeadingTag {...titleProps} className='fd-action-bar__title'>{title}</HeadingTag>
+                {actions && (
+                    <div {...actionProps} className={actionBarActionsClasses}>{actions}</div>
+                )}
+            </div>
+            {description &&
+                <p
+                    {...descriptionProps}
+                    className={actionBarDescriptionClasses}>{description}</p>
+            }
         </div>
     );
 });
 
 ActionBar.displayName = 'ActionBar';
 
 ActionBar.propTypes = {
-    children: PropTypes.node,
+    title: PropTypes.string.isRequired,
+    actionClassNames: PropTypes.string,
+    actionProps: PropTypes.object,
+    actions: PropTypes.node,
+    buttonContainerClassNames: PropTypes.string,
+    buttonProps: PropTypes.object,
     className: PropTypes.string,
-    disableStyles: PropTypes.bool
+    description: PropTypes.string,
+    descriptionProps: PropTypes.object,
+    disableStyles: PropTypes.bool,
+    headingLevel: CustomPropTypes.range(1, 6),
+    titleProps: PropTypes.object,
+    onClick: PropTypes.func
+};
+
+ActionBar.defaultProps = {
+    headingLevel: 3
+};
+
+ActionBar.propDescriptions = {
+    actionClassNames: 'Classnames to spread to the action Button container.',
+    actionProps: 'Props to spread to the action Button container',
+    actions: 'Button components to add to the ActionBar.',
+    buttonContainerClassNames: 'Classnames to spread to the back Button container.',
+    description: 'Localized text for the description.',
+    descriptionProps: 'Additional props to be spread to the description\'s `<p>` element.',
+    headingLevel: 'Heading level. `<h1>` is reserved for the page title.',
+    onClick: 'Callback to pass to the back Button container.'
```

Comment:
isn't it passed to the button and not its container?

Reply:
There is no subcomponent anymore.  That said, should we name this prop something to reflect that it's for the back button (_e.g._ `onBackClick`)?